docs/conf.py

```diff
@@ -10,6 +10,21 @@
 
 import pkg_resources
 
+from pudl.metadata.classes import Package
+from pudl.metadata.resources import RESOURCE_METADATA
+
+# Generate a data dictionary for the documentation:
+# Doing this here means we don't have track script output with Git:
+skip_names = ["datasets", "accumulated_depreciation_ferc1"]
+names = [name for name in sorted(RESOURCE_METADATA) if name not in skip_names]
+package = Package.from_resource_ids(names)
+# Sort fields within each resource by name:
+for resource in package.resources:
+    resource.schema.fields = sorted(
+        resource.schema.fields, key=lambda x: x.name
+    )
+package.to_rst(path="data_dictionaries/pudl_db.rst")
```

Comment:
I suspect that there's a better way to integrate this into the workflow of the docs build. Define a function here and register it with one of the Sphinx build / plugin hooks. Rather than just having it run on import of `conf.py`

Reply:
I think this is clear and effective. Are there drawbacks to this solution?

Reply:
The issue I foresee is that this has substantial external side-effects on import -- any time this module gets imported, the data dictionary RST file gets regenerated. Which is definitely a no-no.  I know that there are several places to hook into the Sphinx build API and if we did that, then this would only happen when the build is actually getting run, and at the appropriate time. The functionality would still be defined here but its execution would be more controlled / isolated.

Reply:
What we want to do is analogous to [this example of deploying static assets into the docs build](https://www.sphinx-doc.org/en/master/development/theming.html#add-your-own-static-files-to-the-build-assets) so I'm just going to go ahead and make the change.